traitlets/tests/test_traitlets.py

```diff
@@ -176,6 +176,28 @@ def _default_x(self):
         self.assertEqual(a.x, 11)
         self.assertEqual(a._trait_values, {'x': 11})
 
+    def test_envvar_dynamic_initializer(self):
+        import os
+
+        os.environ['MY_ENVVAR'] = 'true'
+
+        class A(HasTraits):
+            b = Bool()
+
+            @envvar_default('b', 'MY_ENVVAR')
+            def _b_from_envvar(self, init):
```

Comment:
I was working from the idea that the handlers should generally accept a 'bunch' object, or nothing, so I chose to include the trait in the 'init' bunch. I couldn't really think of a better name than 'init' so if there's something better I'm open to it.